### ISSUE TITLE
Build Not working

### ISSUE THREAD

**Issue by cfalzone** [kindlyops/lambda-clamav-layer#18]
When trying to build this, `yumdownloader` is not able to find any of the packages:
```
57 packages excluded due to repository priority protections
No Match for argument clamav
No Match for argument clamav-lib
No Match for argument clamav-update
No Match for argument json-c
No Match for argument pcre2
No Match for argument libxml2
No Match for argument bzip2-libs
No Match for argument libtool-ltdl
No Match for argument xz-libs
No Match for argument libprelude
No Match for argument gnutls
No Match for argument nettle
No Match for argument libcurl
No Match for argument libnghttp2
No Match for argument libidn2
No Match for argument libssh2
No Match for argument openldap
No Match for argument libffi
No Match for argument krb5-libs
No Match for argument keyutils-libs
No Match for argument libunistring
No Match for argument cyrus-sasl-lib
No Match for argument nss
No Match for argument nspr
No Match for argument libselinux
No Match for argument openssl-libs
No Match for argument libcrypt
Nothing to download
```

Any thoughts?

**Comment by statik** [comment]
hi @cfalzone, thanks for reporting this! I haven't been working with this lately but would absolutely welcome any help you can provide in the form of troubleshooting or even a PR.

**Comment by cfalzone** [comment]
So far all I have come to is that if I take out -- `--archlist=x86_64` it gets through creating the zip, but when I try to use the layer In my function it is giving this error: 
```
clamscan: error while loading shared libraries: libjson-c.so.2: cannot open shared object file: No such file or directory
```

I checked in the zip and it is in there in the lib folder so I don't know how to proceed.

My code to call it using child_proccess.spawnSync in nodejs:
```
    const scan = spawnSync('clamscan', ['', `/tmp/${fileName}`], {
      // cwd: '/tmp',
      encoding: 'utf8',
      stdio: 'inherit',
    });
```

**Comment by ankon** [comment]
I tried reproducing this, but couldn't. Do you have the complete log output of `make build`?

Here's my output for comparison:
[build-20211209-main.log](https://github.com/kindlyops/lambda-clamav-layer/files/7683469/build-20211209-main.log)

Image version used:
~~~
REPOSITORY                                                   TAG         IMAGE ID      CREATED       SIZE
docker.io/library/amazonlinux                                2           d85ab0980c91  4 months ago  170 MB
~~~

I also tried with a newer image, still works:
~~~
REPOSITORY                                                   TAG         IMAGE ID      CREATED       SIZE
docker.io/library/amazonlinux                                2           f9d4a99f2542  6 days ago    171 MB
~~~

**Comment by cfalzone** [comment]
Here is my output -- https://gist.github.com/cfalzone/ea88463d3efc6b939cfa677d6734d8b9

Could this have something to do with building this on an M1 Mac?  My thought was that Docker is running this in a VM so it shouldn't matter.  That's the whole point of running docker right?

**Comment by cfalzone** [comment]
I do seem to have a different Image id

```
REPOSITORY                       TAG       IMAGE ID       CREATED         SIZE
amazonlinux                      2         5fbac2e4cd8c   6 days ago      193MB
```

**Comment by cfalzone** [comment]
That was, at least in part, the issue.
adding: `--platform=linux/x86-64` to the docker run command forces it to use the correct image.

So now I have the layer built and I deployed it with serverless.  But I still get this error when running it:

```
2021-12-09T12:59:35.059000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 LibClamAV Error: cl_load(): No such file or directory: /var/lib/clamav
2021-12-09T12:59:35.059000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 ERROR: Can't get file status
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 ----------- SCAN SUMMARY -----------
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Known viruses: 0
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Engine version: 0.103.3
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Scanned directories: 0
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Scanned files: 0
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Infected files: 0
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Data scanned: 0.00 MB
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Data read: 0.00 MB (ratio 0.00:1)
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Time: 0.038 sec (0 m 0 s)
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 Start Date: 2021:12:09 12:59:35
2021-12-09T12:59:35.060000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 End Date:   2021:12:09 12:59:35
2021-12-09T12:59:35.064000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 2021-12-09T12:59:35.064Z	196fc5e6-934f-5a69-977a-9fcd6ca2a1ee	INFO	{
  message: 'File scan status',
  scanStatus: 2,
  fileId: '0d136b97-cced-4784-a000-8969df4ad67d',
  key: 'processing/0d136b97-cced-4784-a000-8969df4ad67d/raw.jpg'
}
2021-12-09T12:59:35.064000+00:00 2021/12/09/[$LATEST]953c55114efd4d9ca5d09aa1329462b8 2021-12-09T12:59:35.064Z	196fc5e6-934f-5a69-977a-9fcd6ca2a1ee	INFO	{
  message: 'File scan results',
  scan: {
    status: 2,
    signal: null,
    output: [ null, null, null ],
    pid: 24,
    stdout: null,
    stderr: null
  },
  fileId: '0d136b97-cced-4784-a000-8969df4ad67d',
  key: 'processing/0d136b97-cced-4784-a000-8969df4ad67d/raw.jpg'
}
```

This is my handler code: (Typescript)
```
  try {
    const scan = spawnSync('clamscan', ['', `/tmp/${fileName}`], {
      encoding: 'utf8',
      stdio: 'inherit',
    });
    console.log({ message: 'File scan status', scanStatus: scan.status, fileId, key });
    console.log({ message: 'File scan results', scan, fileId, key });
    return true;
  } catch (err) {
    console.log({ message: 'Unable to scan file or file infected.', err, fileId, key });
    return false;
  }
}
```

**Comment by cfalzone** [comment]
Okay I figured it all out.  I did end up get this working. 

1. If you are using an M1 Mac you have pass the `--platform=linux/x86-64` and ensure that the resulting zip file for your layer is built in the docker image not on your mac.  

2.  The error `LibClamAV Error: cl_load(): No such file or directory: /var/lib/clamav` is simply because I never downloaded the virus definitions using freshclam before trying to scan a file.  I ended up making a separate lambda function for this that runs on a 3hr schedule.  

That being said, this was linked in another ticket, I ended up switching to use lambci's yumda repo instead of the process outlined here.  See:  https://github.com/lambci/yumda/issues/3, much simpler build process for my chain.

Thanks for your help.
